### PR TITLE
Lenwe

### PR DESCRIPTION
**Pull request by KrystianD**
I think it's high time to merge this into master

### CODE REVIEW

**Comment by bambuchaAdm** [comment]
Please make it merge without conflict.

File: auth.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+bool authCheckLocal()
+{
+	for (int i = 0; i < numOfCards; ++i)
+		if (auth_compareToStoredCard(i))
+			return true; 
+	return false;
+}
+
+// private implementation
+bool auth_compareToStoredCard(int i)
```
Comment (bambuchaAdm) [inline auth.h:22]: please use static here

File: config.h.example [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#define STASZEK_MODE
+//#define DEBUG
+
+const int toneAccepted = 1260; //Hz
+const int toneRejected = 440;  //Hz
+const int toneDuration = 100;  // milliseconds 
+
+const int lockTransitionTime = 2000 - 300; // in milliseconds
+
+const int debounceDelay = 50; // milliseconds
```
Comment (bambuchaAdm) [inline config.h.example:19]: It is real this value ?

File: config.h.example [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+const int remoteDelay = 2000;
+
+const int servoLockAngle = 0;
+const int servoUnlockAngle = 180;
+
+const int timeBetweenFrames = 50;
+
+const uint8_t mac[] = { 0xDE, 0xAD, 0xBE, 0xDD, 0xFE, 0xED };
+const IPAddress ip(192,168,100,9);
+const IPAddress srvIp(192,168,100,5);
```
Comment (bambuchaAdm) [inline config.h.example:29]: please don't use abbreviation in variable names

File: config.h.example [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+When using Arduino, connect:
+- RX0:	RFID reader TX
+- D2:	reed switch; second side to ground
+- D3:	microswitch; second side to ground
+- D9:	servo (the yellow cable...)
+- D10:	piezo/speaker; second side to ground
+*/
+
+const int pinServo = 5;
+const int pinPiezo = 6;
```
Comment (bambuchaAdm) [inline config.h.example:43]: Please synchronized documentation with constance. Or just confirm it's good value.

File: config.h.example [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+const int pinButtonSwitch = 2;
+const int pinReedSwitch = 3;
+
+const int tamperProtectionSwitch = 10;
```
Comment (bambuchaAdm) [inline config.h.example:48]: Include it connection documentation.

File: config.h.example (hunk 4 above)
Comment (jkramarz) [inline config.h.example:43]: Łukasz, przerzuć się może na Polski. Nie mam pojęcia co napisałeś.

Comment (bambuchaAdm) [inline config.h.example:43]: Jest użyte 6 w komentarzu wyżej jest napisane że D10. Synchronizujcie lub napiszcie że jest dobrze.

File: lock.cpp [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+Servo servo;
+
+bool isDoorLocked = true; //assumed state of door lock on uC power on
+int doorServerRevertTimeout = 0;
+
+int lockTransitionTimeTimeout = 0;
+
+void servoDoTime(int angle, int time);
+
+void lockInit()
```
Comment (bambuchaAdm) [inline lock.cpp:16]: YAGNI

File: lock.cpp [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+{
+}
+void lockProcess()
```
Comment (bambuchaAdm) [inline lock.cpp:19]: YAGNI

File: lock.cpp [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+{
+}
+void lockEvent1MS()
```
Comment (bambuchaAdm) [inline lock.cpp:22]: Don't you think that better name could contains `tick` ?

File: lock.cpp [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+	isDoorLocked = false;
+}
+void lockDoor()
+{
+	if (isDoorLocked)
+		return;
+
+	int timeLeft = lockTransitionTime;
+
+	// if within unlocking process, calculate time to revert process
```
Comment (bambuchaAdm) [inline lock.cpp:62]: DRY with previous function. Should be moved to another function.

**Comment by bambuchaAdm** [comment]
Dokończę resztę review jutro.

File: lock.cpp (hunk 9 above)
Comment (KrystianD) [inline lock.cpp:62]: I don't see nice way to do this, they are quite different.

**Comment by KrystianD** [comment]
OK I've commited those changes, I will make a new PR after tests, or should I do this now? I don't know how to remove conflicts without merging.

File: lock.cpp [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+	if (lockTransitionTimeTimeout)
+		timeLeft = lockTransitionTime - lockTransitionTimeTimeout + 200;
```
Comment (bambuchaAdm) [inline lock.cpp:64]: Magic number

Comment (KrystianD) [inline lock.cpp:64]: Because it's doing magical thing :) I've added a comment.

**Comment by bambuchaAdm** [comment]
The fix are on kdredactor branch in #3 PR

**Pull request closed without merging**